engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -310,7 +310,13 @@ public static void main(String[] args) throws Throwable
         stramChild.heartbeatLoop();
         exitStatus = 0;
       } finally {
-        stramChild.teardown();
+        try {
+          stramChild.teardown();
+        } catch (Exception ex) {
+          if (exitStatus == 0) {
```

Comment:
IMO, that is confusing. If an exception is thrown during teardown(), if exitStatus is not 0, the exception is swallowed, making it hard to debug JVM termination. If it is 0, it leads to JVM exiting with exit status 0, but logging exception both to AM and the container log.